src/raps/actions/crosschainSwap.ts

```diff
@@ -64,18 +59,13 @@ export const executeCrosschainSwap = async ({
   if (flashbots && network === Network.mainnet) {
     logger.debug('flashbots provider being set on mainnet');
     provider = await getFlashbotsProvider();
-  } else {
-    logger.debug('normal provider being set', network);
-    provider = await getProviderForNetwork(network);
-  }
 
-  if (!walletToUse) {
-    walletToUse = await loadWallet(undefined, true, provider);
-  } else {
-    walletToUse = new Wallet(walletToUse.privateKey, provider);
+    // TODO(skylarbarrera): need to check if ledger and handle differently here
+    walletToUse = new Wallet((walletToUse as Wallet).privateKey, provider);
```

Comment:
all of this logic was added so we can init a new wallet that is using flashbots RPC. there is no need for us to reinit a wallet otherwise